src/components/ElementResizeListener.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {
+  useCallback, useRef, useEffect, RefObject,
+} from 'react';
+
+interface ElementResizeListenerProps {
+  onResize: (event: Event) => void;
+}
+
+const ElementResizeListener = ({ onResize }: ElementResizeListenerProps) => {
+  const rafRef = useRef(0);
+  const objectRef: RefObject<HTMLObjectElement> = useRef(null);
+  const onResizeRef = useRef(onResize);
+
+  onResizeRef.current = onResize;
+
+  const localOnResize = useCallback((e: Event) => {
+    if (rafRef.current) {
+      cancelAnimationFrame(rafRef.current);
+    }
+    rafRef.current = requestAnimationFrame(() => {
+      onResizeRef.current(e);
+    });
+  }, []);
+
+  useEffect(() => {
+    const obj = objectRef.current;
+    if (obj?.contentDocument?.defaultView) {
+      obj.contentDocument.defaultView.addEventListener('resize', localOnResize);
+    }
```

Comment:
Remove the if-statement completely and replace by:
```
obj?.contentDocument?.defaultView?.addEventListener('resize', localOnResize);
```
It won't call `addEventListener` unless all sub props exist, so the effect is the same, just shorter.